### PR TITLE
[Merged by Bors] - feat: add lemma `CanonicallyOrderedCommMonoid.single_le_prod`

### PR DESCRIPTION
**Pull request by ocfnash**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Order.lean [hunk 1]
```diff
@@ -417,6 +417,17 @@ theorem prod_mono_set' (f : ι → M) : Monotone fun s ↦ ∏ x in s, f x := fu
 #align finset.prod_mono_set' Finset.prod_mono_set'
 #align finset.sum_mono_set Finset.sum_mono_set
 
+@[to_additive]
+lemma apply_le_prod_of_mem (i : ι) (hi : i ∈ s) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/BigOperators/Order.lean:421]: Do you want both of those arguments to be explicit?

Comment (eric-wieser) [inline Mathlib/Algebra/BigOperators/Order.lean:421]: Arguably this should be renamed to be consistent with [`Finset.single_le_sum`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/BigOperators/Order.html#Finset.single_le_sum)

Comment (ocfnash) [reply Mathlib/Algebra/BigOperators/Order.lean]: Good point; I have made `i` implicit.

Comment (ocfnash) [reply Mathlib/Algebra/BigOperators/Order.lean]: Agreed, that seems appropriate for discoverability. I have proposed some alternative names in https://github.com/leanprover-community/mathlib4/pull/11383/commits/8207488b12655c4bfb7cacf7feb0100f06784343

File: Mathlib/Algebra/BigOperators/Order.lean [hunk 2]
```diff
@@ -417,6 +417,22 @@ theorem prod_mono_set' (f : ι → M) : Monotone fun s ↦ ∏ x in s, f x := fu
...
+@[to_additive "In a canonically-ordered additive monoid, a sum bounds of its terms.
+
+See also `Finset.single_le_sum`."]
+lemma single_le_prod_of_canonicallyOrdered {i : ι} (hi : i ∈ s) :
+    f i ≤ ∏ j in s, f j := by
+  rw [← prod_singleton f i]
+  exact prod_mono_set' _ <| by simp [hi]
+
+@[to_additive]
+lemma single_le_prod_of_canonicallyOrdered_of_fintype [Fintype ι] (i : ι) :
```
Comment (eric-wieser) [inline Mathlib/Algebra/BigOperators/Order.lean:431]:
```suggestion
lemma _root_.Fintype.single_le_prod_of_canonicallyOrdered [Fintype ι] (i : ι) :
```
To match `Fintype.sum_eq_single`

**Comment by eric-wieser** [comment]
@YaelDillies, I think you recently cleaned up some names around here: does this look good to you?

**Review by YaelDillies — COMMENTED** [review]
I did not rename these lemmas recently, no. All I can say is that suffixing with `of_canonicallyOrdered` is not currently a naming convention. The current convention is to put in the `CanonicallyOrderedSomething` namespace.

The PR title is outdated.

File: Mathlib/Algebra/BigOperators/Order.lean [hunk 3]
```diff
@@ -417,6 +417,21 @@ theorem prod_mono_set' (f : ι → M) : Monotone fun s ↦ ∏ x in s, f x := fu
...
+
+See also `Finset.single_le_sum`."]
+lemma single_le_prod_of_canonicallyOrdered {i : ι} (hi : i ∈ s) :
+    f i ≤ ∏ j in s, f j :=
+  single_le_prod' (fun _ _ ↦ one_le _) hi
+
+@[to_additive]
+lemma _root_.Fintype.single_le_prod_of_canonicallyOrdered [Fintype ι] (i : ι) :
+    f i ≤ ∏ j, f j :=
+  Finset.single_le_prod_of_canonicallyOrdered (mem_univ i)
```
[the remark this answers is not in the document]
Comment (ocfnash) [reply Mathlib/Algebra/BigOperators/Order.lean]: Fair enough, I've dropped it.

**Comment by ocfnash** [comment]
> suffixing with `of_canonicallyOrdered` is not currently a naming convention. The current convention is to put in the `CanonicallyOrderedSomething` namespace.

I have renamed the lemma to `CanonicallyOrderedCommMonoid.single_le_prod`. I don't have strong feels but I think I still prefer `Finset.single_le_prod_of_canonicallyOrdered` since:

- the related lemmas directly below all live in the `Finset` namespace
- using the suffix `_of_canonicallyOrdered` is generic over the `CanonicallyOrderedCommMonoid`, `CanonicallyOrderedCommSemiring`, ...

**Comment by YaelDillies** [comment]
I agree, but if you want to change the convention it should be a separate PR.

At any rate, this PR is good now, so

maintainer merge

**Pull request closed without merging**